smart_open/tests/test_smart_open.py

```diff
@@ -785,20 +787,23 @@ def write_read_assertion(self, test_file):
     def test_open_gz(self):
         """Can open gzip?"""
         fpath = os.path.join(CURR_DIR, 'test_data/crlf_at_1k_boundary.warc.gz')
-        data = smart_open.smart_open(fpath).read()
+        with smart_open.smart_open(fpath) as file:
+            data = file.read()
         m = hashlib.md5(data)
         assert m.hexdigest() == '18473e60f8c7c98d29d65bf805736a0d', \
             'Failed to read gzip'
 
     def test_write_read_gz(self):
         """Can write and read gzip?"""
-        test_file = tempfile.NamedTemporaryFile('wb', suffix='.gz', delete=False).name
-        self.write_read_assertion(test_file)
+        with tempfile.NamedTemporaryFile('wb', suffix='.gz', delete=False) as file:
+            test_file_name = file.name
+        self.write_read_assertion(test_file_name)
 
     def test_write_read_bz2(self):
         """Can write and read bz2?"""
-        test_file = tempfile.NamedTemporaryFile('wb', suffix='.bz2', delete=False).name
-        self.write_read_assertion(test_file)
+        with tempfile.NamedTemporaryFile('wb', suffix='.bz2', delete=False) as file:
```

Comment:
small nitpick - can you rename `file` (because this is type in python), for example, use `infile` (here and anywhere)